deploy-agent/deployd/client/client.py

```diff
@@ -28,14 +29,33 @@
 
 class Client(object):
     def __init__(self, config=None, hostname=None,
-                 ip=None, hostgroup=None, host_id=None):
+                 ip=None, hostgroup=None, host_id=None, use_facter=None):
         self._hostname = hostname
         self._ip = ip
         self._hostgroup = hostgroup
         self._id = host_id
         self._config = config
+        self._use_facter = use_facter
 
     def _read_host_info(self):
+        if self._use_facter:
+            if not self._hostname and self._config.get_facter_name_key():
+                self._hostname = utils.get_info_from_facter(self._config.get_facter_name_key())
+
+            if not self._ip and self._config.get_facter_ip_key():
+                self._ip = utils.get_info_from_facter(self._config.get_facter_ip_key())
+
+            if not self._id and self._config.get_facter_id_key():
+                output = utils.get_info_from_facter(self._config.get_facter_id_key())
+                if output:
+                    self._id = output.split(",")
+
+            if not self._hostgroup and self._config.get_facter_group_key():
+                output = utils.get_info_from_facter(self._config.get_facter_group_key())
+                if output:
+                    self._hostgroup = output.split(",")
+
+        # read host_info file
```

Comment:
Let us make the host_info inside an else block - if we use factor, we should not parse host_info anymore